### PR TITLE
Updated to README and docs. Simplified sampling_tests.jl by using new…

### PR DESCRIPTION
**Pull request by goedman**
This PR contains some finishing touches related to the sampling-api (now using the Array constructor feature) and documentations updates and added a few sections to the README.

Cameron, I did get warnings on the additional convert() method in constructors.jl, basically pointing out it was overwriting chains.jl line 338. So in this PR I removed it. I also moved the saving and loading of chains towards the end of the README, just above the export and sample sections.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -152,18 +152,6 @@ You can access each of the `P[. . .]` variables by indexing, using `x.P[1]`, `x.
 
 Note that `x.P` is a tuple which has to be indexed by the relevant index, while `x.D` is just a vector.
 
-### Saving and Loading Chains
```
Comment (cpfiffer) [inline README.md]: Why remove this? I'm sure someone would get some use out of this.

Comment (cpfiffer) [reply README.md]: Nevermind, I see that it was just scooted down a bit.

File: src/constructors.jl [hunk 2]
```diff
@@ -120,8 +120,6 @@ function Array(chn::MCMCChains.AbstractChains,
   b
 end
 
-Base.convert(::Type{Array}, chn::MCMCChains.Chains) = convert(Array, chn.value)
```
Comment (cpfiffer) [inline src/constructors.jl]: Good catch, thank you!

**Comment by cpfiffer** [comment]
I merged this over as it was a pretty straightforward PR. Thanks, Rob! Can't help but feel like you're trying to make me redundant around here.

**Comment by goedman** [comment]
Not at all, could never have gotten here without your help and you taking the lead.

The big missing part is the NamedTuple stuff (and consequently preserving column types). Ideally Turing would produce those as Turing knows these types and the fact that a parameter is really an array or even more complex. 

And after the df-based chain summary I really need to go back to StatisticalRethinking & friends.

Should we make a new release? Should probably be a minor update release (at least that how I usually do it in CmdStan & SR) as quite a bit new functionality is there.

**Comment by cpfiffer** [comment]
Sure, I'll get one out soon.

**Comment by cpfiffer** [comment]
The release is headed out. As you go back to StatisticalRethinking and find anything you need in particular from this repo, make sure to open an issue and I'll try to make sure I can work on. I appreciate all your help here!

**Comment by goedman** [comment]
Yes, this morning it showed up. I wonder if there is a delay between merging into METADATA and the new Registry.

**Comment by cpfiffer** [comment]
Strange, that hasn't been my experience. Usually if I call `] up`  the new package shows up pretty quick after the merge.

**Comment by goedman** [comment]
Yip, same here, but the last week or so it feels different. I don't TuringLang is ever using Project.toml etc. But not a big deal, in the end it worked.

**Pull request merged**